src/sentry/models/counter.py

```diff
@@ -103,32 +104,37 @@ def create_counter_function(sender, using, **kwargs):
         return
 
     cursor = connections[using].cursor()
-    cursor.execute(
-        '''
-        create or replace function sentry_increment_project_counter(
-            project bigint, delta int) returns int as $$
-        declare
-          new_val int;
-        begin
-          loop
-            update sentry_projectcounter set value = value + delta
-             where project_id = project
-               returning value into new_val;
-            if found then
-              return new_val;
-            end if;
+    try:
```

Comment:
Yes. I'd like to get rid of it. They use it I believe to count the projects on the DB end for performance reasons (I'm sure).

I believe they use it since they have a huge instance in the cloud and probably an enormous amount of projects. We could easily just count the projects on the server side when required. I've added a TODO to fix it (below).

Reply:
... in general I believe we should not use any DB functions in the end product, as having the business logic only in the server makes it much easier to test.